eZ/Bundle/EzPublishCoreBundle/Imagine/Cache/AliasGeneratorDecorator.php

```diff
@@ -41,14 +41,12 @@ class AliasGeneratorDecorator implements VariationHandler
     /**
      * @param \eZ\Publish\SPI\Variation\VariationHandler $aliasGenerator
      * @param \Psr\Cache\CacheItemPoolInterface $cache
-     * @param \eZ\Publish\Core\MVC\Symfony\SiteAccess $siteAccess
      * @param \Symfony\Component\Routing\RequestContext $requestContext
      */
-    public function __construct(VariationHandler $aliasGenerator, CacheItemPoolInterface $cache, SiteAccess $siteAccess, RequestContext $requestContext)
+    public function __construct(VariationHandler $aliasGenerator, CacheItemPoolInterface $cache, RequestContext $requestContext)
```

Comment:
Maybe keep `SiteAccess` as the last optional argument so it's gonna be possible to inject it in the future.

The fact we have to use setter is not a normal situation IMHO, we probably should look for the solution from a different angle, because races in container builder related to so called "SiteAccess hell", have been hitting us hard recently. // cc @andrerom 

That said, due to Sf architecture some things might indeed not be possible, but I need to investigate this ;)

Reply:
Alternative is SiteAccess stack object, ref RequestStack which Symfony ended up with for similar problem for Request object. Config Resolver is such an object.

Reply:
Actually disregard my request to keep it in a constructor. If we're gonna fix it, it's probably gonna be some other service, e.g. `SiteAccessStack` (following Symfony convention).

Reply:
thats just for siteaccess object, same issue exists for all SA aware settings as implied above.